internal/progress/reporter.go

```diff
@@ -35,12 +51,13 @@ func PushMsg(update v1.Update) string {
 	return fmt.Sprintf("Uploaded: %.2f MB", float64(update.Complete)/1024/1024)
 }
 
-func NewProgressReporter(w io.Writer, msgF progressF) *Reporter {
+func NewProgressReporter(w io.Writer, msgF progressF, layer v1.Layer) *Reporter {
 	return &Reporter{
 		out:      w,
-		progress: make(chan v1.Update),
+		progress: make(chan v1.Update, 100),
```

Comment:
I don't think there's any value in buffering this beyond 1 element - any buffer size will eventually start transmitting backpressure if there's no mechanism for dropping intermediate updates that aren't serviced in a certain period of time (i.e. if the producer is producing faster than the consumer is consuming, you'll always fill the buffer in time, and then you'll be back to blocking again, and the only way to avoid that is to drop intermediate updates).  I think the only pattern where buffering > 1 really makes sense is when using the channel as a semaphore.  I would buffer 1 element and add either a `default:` case or a timeout case on the channel write.

Reply:
Thanks for the explanation, I didn’t properly understand it before.. 
I have added your suggestion of dropping intermediate updates here: [a818eab](https://github.com/docker/model-distribution/pull/80/commits/a818eabd9a64290e74051fc4c63667bd1097d0e4)